openpgp/s2k/s2k.go

```diff
@@ -168,11 +175,26 @@ func parseGNUExtensions(r io.Reader) (f func(out, in []byte), err error) {
 		return
 	}
 	gnuExtType := int(buf[0])
-	if gnuExtType != 1 {
-		return nil, errors.UnsupportedError("unknown S2K GNU protection mode: " + strconv.Itoa(int(gnuExtType)))
-	}
+	if gnuExtType == 1 {
+		return nil, nil
+	} else if gnuExtType == 2 {
+		var lenBuf [1]byte
+		_, err = io.ReadFull(r, buf[:1])
+		if err != nil {
+			return
+		}
+
+		var ivBuf [16]byte
+		ivLen := minInt(lenBuf[0], 16)
```

Comment:
No, I had that backwards: http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=g10/parse-packet.c;h=c925e9440e371a223f90ac71d7379b2bfc706c40;hb=refs/heads/STABLE-BRANCH-2-0#l1993